backtracking/all_permutations.py

```diff
@@ -2,7 +2,7 @@
 	In this problem, we want to determine all possible permutations
 	of the given sequence. We use backtracking to solve this problem.
 
-	Time complexity: O(n!),
+	Time complexity: O(n! * n),
```

Comment:
Why the complexity is `O(n! * n)`?

Reply:
At least, we print the permutations. Each permutation size is n.

Reply:
Fixing a character at the 0th index, recursion will run n! times. And this will be repeated for every character of the string. So O(n! * n). Makes sense?

Reply:
For the last character, I mean the (n-1)th one, we iterate through numbers from 0 to n-1.
So for each n! states, our for loop takes O(n) time.

Reply:
I'd agree on this. But what about generating subsequences? We don't have any loops there.

Reply:
I guess you're right.
T(n) = 2 * T(n - 1) + 1, so T(n) equals 2^n - 1.
But I think it's better to write O(2^n + n), because we fill the answer list.
And one another thing: how can I commit only one file changes?

Reply:
Anyway, O(2^n + n) = O(2^n)